include/deal.II/lac/sparsity_pattern_base.h

```diff
@@ -106,6 +108,17 @@ class SparsityPatternBase : public Subscriptor
   add_entries(const ArrayView<const size_type> &rows,
               const ArrayView<const size_type> &columns);
 
+  /**
+   * General function for adding new entries from an unsorted list of pairs.
+   *
+   * This function is useful when multiple entries need to be added which do
+   * not correspond to a particular row, e.g., when assembling a flux sparsity
+   * pattern.
+   */
+  virtual void
+  sort_and_add_entries(
+    const ArrayView<const std::pair<size_type, size_type>> &entries);
```

Comment:
I don't want to force you through contortions but think that what the function does internally shouldn't be reflected in the function's name. For example, you (correctly) don't mention the sorting in the documentation either.

Why would you have to add the `using` directive to the derived classes? Because they declare their own `add_entries()` functions and you want to expose the function of same name declared in the base class?

Reply:
I thought about it some more and I think renaming the function is the right choice.

re: using statements: yes (they each have a version templated on the iterator type)